clients/java/signalr/src/main/java/HandshakeProtocol.java

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import com.google.gson.Gson;
+
+public class HandshakeProtocol {
+    public static Gson gson = new Gson();
+    private static final String RECORD_SEPARATOR = "\u001e";
+
+    public static String createHandshakeRequestMessage(HandshakeRequestMessage message){
+        // The handshake request is always in the JSON format
+        return gson.toJson(message) + RECORD_SEPARATOR;
+    }
```

Comment:
We should also add parsing logic to parse a `String` into `HandshakeResponseMessage`

Reply:
Added a `parseHandshakeResponse` method here. Adding unit tests for the HandshakeProtocol now